TakeScoop/typhoon
# Pass cloud_provider var to bootkube module

High level description of the change.

* Allow cloud_provider variable to pass through typhoon to the bootkube render

## Testing

Testing will be to modify the api server and controller manager arguments in place to use the `--cloud-provider="aws"` flag. We should also recreate playground with this change to test that a new cluster comes up as expected, however in place modification will be how the changes are rolled out to staging and production.

The behavior we really want to test is that the kube-controller-manager is able to connect to AWS and create a load balancer given the proper service annotations.

This pr wont be merged until all of the following have been tested in playground:
- kubernetes repo, pass cloud_provider: https://github.com/TakeScoop/kubernetes/pull/400
- typhoon repo, pass cloud_provider: https://github.com/TakeScoop/typhoon/pull/29
- Update IAM permissions for controller instance profile to allow load balancing and DNS record creation (WIP) https://github.com/TakeScoop/typhoon/pull/30
- update external-dns add-on to create a well defined DNS record for services with the appropriate service annotation https://github.com/TakeScoop/kubernetes/pull/401
